common/src/main/java/com/itsaky/androidide/utils/FileUtils.kt

```
@@ -0,0 +1,112 @@
+		decoder.onMalformedInput(CodingErrorAction.REPORT)
+		decoder.onUnmappableCharacter(CodingErrorAction.REPORT)
+
+		return try {
+			val header = ByteArray(UTF8_SNIFF_LENGTH)
+			val read = file.inputStream().use { it.read(header) }
+			if (read <= 0) {
+				return false
+			}
+			decoder.decode(ByteBuffer.wrap(header, 0, read))
+			true
+		} catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)